### ISSUE TITLE
Use meteor CLI from the same root directory, npm run meteor -- ...

### ISSUE THREAD

**Issue by grigio** [jedwards1211/meteor-webpack-react#14]
Sometimes you may want to manage the `meteor_app` or `meteor_core` via meteor command.. but it isn't available outside it's directory.

So instead change directory and running

```
meteor <arguments>
meteor search babel
```
you can do from the usual directory:
```
❯ npm run meteor search babel ⏎ # or npm run meteor -- --version

> meteor-webpack-react@0.0.1 meteor /Users/grigio/+persistant/Code/testing/meteor-webpack-react/webpack
> cd ../meteor && meteor "search" "babel"

Matching packages:                            
babel-compiler             Parser/transpiler for ECMAScript 2015+ syntax
babel-runtime              Runtime support for output of B
```

**Comment by jedwards1211** [comment]
I'm a bit confused, do you mean you added an npm script that just does `cd ../meteor && meteor` followed by the script arguments?

**Comment by grigio** [comment]
Yes, some commands may need `--` but I think it is useful to avoid multiple terminal open in different paths for the same app. What do you think?

**Comment by jedwards1211** [comment]
hmm.  We could put a script in that directory so that the user can just go `./meteor do something` but I can't think of any way to make it work without the `./`...can you?

**Comment by AdamBrodzinski** [comment]
You can't unless you add `.` to the path but that would be bad security wise. I also think running `meteor search ...` would be a bit confusing if they needed to actually edit the `meteor_core` files.

When you run `run npm command` does it allow you to accept arguments? like `npm run meteor search babel` ?

**Comment by jedwards1211** [comment]
I'm sure it does, though personally I'd rather just put a script named `meteor` in the root directory so I can `./meteor ...`

**Comment by AdamBrodzinski** [comment]
yea I think that's best actually... it's shorter and more obvious what's going on. You could even make an alias like `wmeteor` for `./meteor`.

**Comment by jedwards1211** [comment]
oh yeah, that would be nice too

**Comment by grigio** [comment]
@AdamBrodzinski yes, `npm run meteor search babel` works. Yes probably a custom alias is shorter
```
alias m="cd meteor && meteor"
~/C/t/meteor-webpack-react ❯❯❯ m --version
Meteor 1.1.0.3
```

**Comment by AdamBrodzinski** [comment]
what about this :smiley: 

`./met`
    >>> starts up webpack  

`./met --version` 
     >>>  Meteor 1.1.0.3

This is a pretty naive script but could be augmented with `--production`  checks to make it smarter.

```
#!/bin/bash

if [ -f $1 ];
then
  eval "cd webpack && npm run dev"
else
  eval "cd meteor_core && meteor $@"
fi
```

**Comment by grigio** [comment]
Yeah probably is better to manage it in that way.
The important thing is use both CLI from the same root directory

**Comment by AdamBrodzinski** [comment]
@jedwards1211 @grigio we can prob. keep this open if we're going to include the `met` script from this PR https://github.com/jedwards1211/meteor-webpack-react/pull/15#event-381637227

**Comment by jedwards1211** [comment]
Yeah, I agree with both of you on this.

Instead of `cd webpack && npm run dev` though it should just be `npm run dev` from the root dir though.
I'm not an expert on shell scripting, so I was wondering, how come you used `eval`?

**Comment by AdamBrodzinski** [comment]
> I'm not an expert on shell scripting, so I was wondering, how come you used eval?

Haha, yea don't need those anymore, they should go... I first started out another direction but left it in.

should we intercept `--build`, `--debug` and `--production` for `$1` and run our commands instead? Seems like it would avoid confusion when they get the wrong result :laughing: 

I'm soooo going to hook into this on my personal fork.... `met generate component PostsList` which would scaffold out a yo generator to spit out the component, sass, test, files :sunglasses:  (where's the lazy emoji lol)

**Comment by jedwards1211** [comment]
I think it's more likely to cause confusion if we intercept anything.  To me it's clearest if `meteor` is `meteor`, `webpack` is `webpack`, and anything else is a custom script.  Since the `met` script blurs those lines I think it's least confusing if it's simply a shortcut for `cd meteor && meteor`.

**Comment by jedwards1211** [comment]
And in line with that, I'd rather `met` by itself not do `npm run dev`.

**Comment by AdamBrodzinski** [comment]
Sounds good to me :+1:  Just pushed PR https://github.com/jedwards1211/meteor-webpack-react/pull/17 to fix this

**Comment by jedwards1211** [comment]
Cool, thanks!  Yeah I don't have to do a whole lot with the meteor command in my own work so maybe I'm not familiar with some common use cases, but I figure if `./met mongo` or whatever is less typing than `cd meteor` followed by `meteor mongo`, anything past that point is splitting hairs.

**Comment by AdamBrodzinski** [comment]
Yep, I mainly just use it to launch the app.... though 90% of the time I have a makefile for launching w/ or w/o tests, linting, and deploy scripts. Now that I think of it.... `meteor mongo` and occasionally `meteor shell` is all I ever use.

Hmm just thought about deploying to Meteor.com 's servers.... I just tried it with `npm run prod` then cancel the watch then `./met deploy webpack-test.meteor.com` and it works just fine :smiley: . I'm thinking that a target/script might be overkill for this but would be an option if there's a lot of issues related to deploying.

Their CLI package search is usually too bloated so I have an Alfred action to search Atmosphere if I need Meteor specific stuff..... however i'm really looking forward to the NPM workflow :sunglasses:

**Comment by grigio** [comment]
fixed in #17
